sbi/inference/posteriors/base_posterior.py

```diff
@@ -355,26 +359,42 @@ def _slice_np_mcmc(
             initial_params: Initial parameters for MCMC chain.
             thin: Thinning (subsampling) factor.
             warmup_steps: Initial number of samples to discard.
+            vectorized: Whether to use a vectorized implementation of
+                the Slice sampler (still experimental).
+            show_progress_bars: Whether to show a progressbar during sampling;
+                can only be turned off for vectorized sampler.
 
         Returns: Tensor of shape (num_samples, shape_of_single_theta).
         """
-
         num_chains = initial_params.shape[0]
         dim_samples = initial_params.shape[1]
 
-        all_samples = []
-        for c in range(num_chains):
-            posterior_sampler = SliceSampler(
-                utils.tensor2numpy(initial_params[c, :]).reshape(-1),
-                lp_f=potential_function,
-                thin=thin,
+        if not vectorized:  # Sample all chains sequentially
+            all_samples = []
+            for c in range(num_chains):
+                posterior_sampler = SliceSampler(
+                    utils.tensor2numpy(initial_params[c, :]).reshape(-1),
+                    lp_f=potential_function,
+                    thin=thin,
+                )
+                if warmup_steps > 0:
+                    posterior_sampler.gen(int(warmup_steps))
+                all_samples.append(posterior_sampler.gen(int(num_samples / num_chains)))
+            all_samples = np.stack(all_samples).astype(np.float32)
+            samples = torch.from_numpy(all_samples)  # chains x samples x dim
+        else:  # Sample all chains at the same time
+            posterior_sampler = SliceSamplerVectorized(
+                init_params=utils.tensor2numpy(initial_params),
+                log_prob_fn=potential_function,
+                num_chains=num_chains,
+                verbose=show_progress_bars,
             )
-            if warmup_steps > 0:
-                posterior_sampler.gen(int(warmup_steps))
-            all_samples.append(posterior_sampler.gen(int(num_samples / num_chains)))
-        all_samples = np.stack(all_samples).astype(np.float32)
-
-        samples = torch.from_numpy(all_samples)  # chains x samples x dim
+            warmup_ = warmup_steps * thin
+            num_samples_ = int((num_samples * thin) / num_chains)
```

Comment:
Shouldn't we use `ceil()` here? If `num_samples*thin` is not a multiple of `num_chains`, it will not produce enough samples, no?